_posts/2024-07-09-datafusion-40.0.0.md

```diff
@@ -0,0 +1,248 @@
+* We published a paper in SIGMOD [Apache Arrow DataFusion: A Fast, Embeddable, Modular Analytic Query Engine](https://dl.acm.org/doi/10.1145/3626246.3653368)
+* Many interesting posts: TODO find some
+* DataFusion mentioned as a candidate to be the POSIX of databases in ["What Goes Around Comes Around... And Around...](https://db.cs.cmu.edu/papers/2024/whatgoesaround-sigmodrec2024.pdf)
+* ["Why you should keep an eye on Apache DataFusion and its community."](https://www.cpard.xyz/posts/datafusion/)
+
+
+Meetups:
+Austin https://github.com/apache/datafusion/discussions/8522
+San Francisco
+Hangzhou
+NYC https://github.com/apache/datafusion/discussions/11213
+Belgrade
+
+[CHANGELOG]: https://github.com/apache/arrow-datafusion/blob/main/datafusion/CHANGELOG.md
+
+# Improved Performance 🚀 
+
+Performance is a key feature of DataFusion, DataFusion is 
+
+TODO: Planning speed improvements: 
+
+Implement specialized group values for single Uft8/LargeUtf8/Binary/LargeBinary column #8827
+
+
+https://github.com/apache/datafusion/pull/8827
+
+## In progress
+StringView (TODO get links)
+
+
+
+# Improved quality
+
+TODO mention the SQLancer
+
+
+# New Features ✨
+
+Improvements: unnest, null handling improvements for lead/lag
+
+## 
+
+## TreeNode API
+
+
+## Other Features
+
+Recursive CTEs https://github.com/apache/datafusion/pull/9619 / https://github.com/apache/datafusion/issues/462
+
+# Building Systems is Easier with DataFusion 🛠️
+
+Large scale "extract scalar functions from the core": all scalar functions are now exactly the same API as built in https://github.com/apache/datafusion/issues/9285
+
+We are close to completing the same treatment for aggregates and then window functions
+
+SQL to String (both exprs and plans): https://github.com/apache/datafusion/issues/9494
+
+WASM builds: https://github.com/apache/datafusion/discussions/9834
+
+## Documentation
+It is easier than ever to get started using DataFusion with the
+new [Library Users Guide] as well as significantly improved the [API documentation]. 
+
+[Library Users Guide]:https://arrow.apache.org/datafusion/library-user-guide/index.html
+[API documentation]: https://docs.rs/datafusion/latest/datafusion/index.html
+
+## User Defined SQL Extensiions
+
+## Support for `CREATE FUNCTION`
+https://github.com/apache/datafusion/pull/9333
+
+Let's you build systems that support user defined functions
+
+Huge thanks to [@milenkovicm](https://github.com/milenkovicm)
+
+```sql
+CREATE FUNCTION my_func(DOUBLE, DOUBLE)
+    RETURNS DOUBLE
+    RETURN $1 $3
+"#;
+```
+
+And
+
+```sql
+CREATE FUNCTION iris(FLOAT[])
+RETURNS FLOAT[]
+LANGUAGE TORCH
+AS 'models:/iris@champion'
+```
+
+```sql
+CREATE FUNCTION func(FLOAT[])
+RETURNS FLOAT[]
+LANGUAGE WASM
+AS 'func.wasm'
+```
+
+BTW it would be great if someone made a demo showing how to do this (see https://github.com/apache/datafusion/issues/9326 )
+
+
+## Parquet indexing / low latency queries
```

Comment:
We used the Parquet indexing feature (`ParquetAccessPlan`) to add [efficient support] for reading from DeltaLake tables and handling [deletion vectors]. I think its a cool use-case for the indexing feature and perhaps worth calling out as an example of a real-world use-case.

[efficient support]: https://github.com/spiceai/spiceai/pull/1891
[deletion vectors]: https://docs.delta.io/latest/delta-deletion-vectors.html